neuralpp/symbolic/interval.py

```diff
@@ -104,10 +108,11 @@ def __iter__(self) -> Iterable[int]:
         raise NotImplementedError("TODO")
 
 
-def from_constraint(index: Variable, constraint: Context) -> Expression:
+def from_constraint(index: Variable, constraint: Context, context: Context, is_integral: bool) -> Expression:
```

Comment:
I believe this should be the correct interface. Instead of `from_constraint(index, constraint & context, is_integral)`,
we should call `from_constraint(index, constraint, context, is_integral)` and provide the context separately. This also enables more test cases given our current interval implementation.

Reply:
Clarifying question: what is the difference between constraint and context?

Reply:
`constraint` is part of the quantifier expression. `context` is "current context in the if-then-else tree" so e.g. in `if A then (if B then C else D) else E`, the context of C is `A & B`.